OptimizelySDK.Tests/AudienceConditionsTests/ConditionEvaluationTest.cs

```diff
@@ -334,5 +373,230 @@ public void TestLTMatcherReturnsTrueWhenAttributeValueIsLessThanConditionValue()
         }
 
         #endregion // LTMatcher Tests
+
+        #region LEMatcher Tests
+        [Test]
+        public void TestLEMatcherReturnsFalseWhenAttributeValueIsGreaterAndTrueIfEqualToConditionValue()
+        {
+            Assert.IsFalse(LECondition.Evaluate(null, new UserAttributes { { "distance_le", 15 } }, Logger)  ?? true);
+            Assert.IsTrue(LECondition.Evaluate(null, new UserAttributes { { "distance_le", 10 } }, Logger) ?? false);
+        }
+
+        [Test]
+        public void TestLEMatcherReturnsNullWhenAttributeValueIsNotANumericValue()
+        {
+            Assert.IsNull(LECondition.Evaluate(null, new UserAttributes { { "distance_le", "invalid_type" } }, Logger));
+            LoggerMock.Verify(l => l.Log(LogLevel.WARN, @"Audience condition {""type"":""custom_attribute"",""match"":""le"",""name"":""distance_le"",""value"":10} evaluated to UNKNOWN because a value of type ""String"" was passed for user attribute ""distance_le""."), Times.Once);
+        }
+
+        [Test]
+        public void TestLEMatcherReturnsNullWhenAttributeValueIsOutOfBounds()
+        {
+            Assert.IsNull(LECondition.Evaluate(null, new UserAttributes { { "distance_le", double.NegativeInfinity } }, Logger));
+            Assert.IsNull(LECondition.Evaluate(null, new UserAttributes { { "distance_le", -Math.Pow(2, 53) - 2 } }, Logger));
+            LoggerMock.Verify(l => l.Log(LogLevel.WARN, @"Audience condition {""type"":""custom_attribute"",""match"":""le"",""name"":""distance_le"",""value"":10} evaluated to UNKNOWN because the number value for user attribute ""distance_le"" is not in the range [-2^53, +2^53]."), Times.Exactly(2));
+        }
+
+        [Test]
+        public void TestLEMatcherReturnsTrueWhenAttributeValueIsLessThanConditionValue()
+        {
+            Assert.IsTrue(LECondition.Evaluate(null, new UserAttributes { { "distance_le", 5 } }, Logger) ?? false);
+        }
+
+        #endregion // LEMatcher Tests
+
+        #region SemVerLTMatcher Tests
```

Comment:
Can we add some invalid attribute tests? (i.e.  "3.0-", 3.x", 3.2, "3.0.0+", "+")?